src/weighted_graph.py

```diff
@@ -0,0 +1,236 @@
+            prev = []
+        if start in prev:
+            return []
+        path_list = [start]
+        nodes = []
+        for each in self._nodes[start]:
+            nodes.append(each[0])
+        prev.append(start)
+        for node in nodes:
+            path_list.extend(self.depth_first_traversal(node, prev))
+        return path_list
+
+    def breadth_first_traversal(self, parent, path_list=None):
+        """Return a list containing the nodes of the graph in order of breadth-first traversal."""
+        if path_list is None:
+            path_list = []
+        if not isinstance(parent, list):
+            path_list.append(parent)
+            parent = [parent]
+        children = []
+        for item in parent:
+            if self._nodes[item]:
+                for edge in self._nodes[item]:
+                    if edge[0] not in path_list:
+                        children.append(edge[0])
+        path_list.extend(children)
+        if len(children) == 0:
+            return path_list
+        return self.breadth_first_traversal(children, path_list)
+
+
+    def djikstras_shortest_path(self, start_node, dest_node):
+        """Return the minimum distance between a start node
+        and a destination node, and the path of minimum distance
+        between those nodes."""
+        unvisited = self.depth_first_traversal(start_node)
+
+        if dest_node not in unvisited:
+            raise ValueError("No path from start node to destination node.")
+
+        distances_and_paths = {i: [None, [None]] for i in unvisited}
+        distances_and_paths[start_node] = [0, [start_node]]
+        current_node = start_node
+
+        while True:
+            for item in self.neighbors(current_node):
+                study_node = item[0]
+                current_node_distance = (distances_and_paths[current_node])[0]
+                potential_distance = item[1] + current_node_distance
+                original_distance = (distances_and_paths[study_node])[0]
+
+                if study_node not in unvisited:
+                    continue
+                elif original_distance is None:
+                    current_node_path = (distances_and_paths[current_node])[1][:]
+
+                    current_node_path.append(item[0])
+
+                    (distances_and_paths[item[0]])[0] = item[1] + current_node_distance
+                    (distances_and_paths[item[0]])[1] = current_node_path
+
+                    continue
+                elif potential_distance < original_distance:
+                    current_node_path = (distances_and_paths[current_node])[1][:]
+
+                    current_node_path.append(item[0])
+
+                    (distances_and_paths[item[0]])[0] = item[1] + current_node_distance
+                    (distances_and_paths[item[0]])[1] = current_node_path
+
+            if current_node == dest_node:
+
+                return str((distances_and_paths[dest_node])[0]) + ' ' + str((distances_and_paths[dest_node])[1])
```

Comment:
Not a fan of this line.  I don't think returning this as a string is proper.  It would be more useful to return this data so that it can actually be used somewhere else, without the sillyness of converting strings back and forth between data types.  In general, avoid returning a string representation of data in a function, unless that is the explicit purpose of that function.  Instead, in this case, return a tuple of the two pieces of data.